wasm/__init__.py

```diff
@@ -4602,45 +4599,62 @@ def type_table(store, tableaddr):
 
 def read_table(store, tableaddr, i):
     if len(store["tables"]) < tableaddr:
-        return "error"
+        raise ValidationError(
+            f"Table address outside of allowed range: {tableaddr} > "
+            f"{len(store['tables'])}"
+        )
     if type(i) != int or i < 0:
-        return "error"
+        raise ValidationError(
+            f"Invalid table index.  Must be positive integer.  Got {repr(i)}"
+        )
     ti = store["tables"][tableaddr]
     if i >= len(ti["elem"]):
-        return "error"
+        raise ValidationError(
+            f"Index out of range for table.  {i} >= {len(ti['elem'])}"
```

Comment:
To match the pattern from the other exceptions, looks like this should say `{i} < {len...`.